src/app/shared/services/backend/link-preview.service.ts

```diff
@@ -0,0 +1,53 @@
+import {Injectable} from '@angular/core'
+import {HttpClient, HttpParams} from '@angular/common/http'
+import {Observable} from 'rxjs'
+import {environment} from '../../../../environments/environment'
+import {map} from "rxjs/operators"
+
+@Injectable({
+  providedIn: 'root',
+})
+export class LinkPreviewService {
+  path = `${environment.backendURL}/api/link`
+
+  constructor(private http: HttpClient) {
+  }
+
+  previewLink(url: string): Observable<LinkPreviewResponse> {
+    const params = new HttpParams().set('url', url)
+    return this.http.get<LinkPreviewResponse>(`${this.path}/preview`, {params: params}).pipe(
+      map((response) => {
+        if (response.open_graph?.image?.url) {
+          return {
+            url: response.url,
+            open_graph: {
+              ...response.open_graph,
+              image: {
+                ...response.open_graph.image,
+                url: response.open_graph.image.url.replace(/&amp;/, '&')
+              }
+            }
+          }
+        }
+        return response
+      })
```

Comment:
Try adding a catchError operator at the end of the service call to return just URL so we can fallback and show it in case of error.